experiments/AtmosGCM/solid-body-rotation.jl

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env julia --project
+using ClimateMachine
+ClimateMachine.init(parse_clargs = true)
+
+using ClimateMachine.Atmos
+using ClimateMachine.Orientations
+using ClimateMachine.ConfigTypes
+using ClimateMachine.Diagnostics
+using ClimateMachine.GenericCallbacks
+using ClimateMachine.ODESolvers
+using ClimateMachine.TurbulenceClosures
+using ClimateMachine.SystemSolvers: ManyColumnLU
+using ClimateMachine.Mesh.Filters
+using ClimateMachine.Mesh.Grids
+using ClimateMachine.TemperatureProfiles
+using ClimateMachine.VariableTemplates
+
+using LinearAlgebra
+using StaticArrays
+using Test
+
+using CLIMAParameters
+using CLIMAParameters.Planet:
+    day, planet_radius
+struct EarthParameterSet <: AbstractEarthParameterSet end
+const param_set = EarthParameterSet()
+
+function init_solid_body_rotation!(problem, bl, state, aux, coords, t)
+    FT = eltype(state)
+
+    ## Assign state variables
+    state.ρ = aux.ref_state.ρ
```

Comment:
It would be nice here to set up an initial state that is hydrostatic but differs from the reference state. For example, you could choose a state from the family of `DecayingTemperatureProfile` but with different parameters than in the ref state. (The ref state should be trivially maintained the way the model is set up, unless it is perturbed by filters. But this is not necessarily the case for other hydrostatic states.)

Reply:
Done.